internal/services/cognitive/cognitive_account_rai_blocklist_resource.go

```diff
@@ -0,0 +1,216 @@
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+type cognitiveRaiBlocklistModel struct {
+	Name               string `tfschema:"name"`
+	CognitiveAccountId string `tfschema:"cognitive_account_id"`
+	Description        string `tfschema:"description"`
+}
+
+type CognitiveRaiBlocklistResource struct{}
+
+var _ sdk.Resource = CognitiveRaiBlocklistResource{}
+
+func (c CognitiveRaiBlocklistResource) Arguments() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"cognitive_account_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: raiblocklists.ValidateAccountID,
+		},
+
+		"description": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+		},
+	}
+}
+
+func (c CognitiveRaiBlocklistResource) Attributes() map[string]*schema.Schema {
+	return map[string]*schema.Schema{}
+}
+
+func (c CognitiveRaiBlocklistResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			var model cognitiveRaiBlocklistModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			client := metadata.Client.Cognitive.RaiBlocklistsClient
+			accountId, err := raiblocklists.ParseAccountID(model.CognitiveAccountId)
+			if err != nil {
+				return err
+			}
+
+			locks.ByID(accountId.ID())
+			defer locks.UnlockByID(accountId.ID())
+
+			id := raiblocklists.NewRaiBlocklistID(accountId.SubscriptionId, accountId.ResourceGroupName, accountId.AccountName, model.Name)
+			existing, err := client.Get(ctx, id)
+			if err != nil && !response.WasNotFound(existing.HttpResponse) {
+				return fmt.Errorf("checking for existing %s: %+v", id, err)
+			}
+
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(c.ResourceType(), id)
+			}
+
+			properties := &raiblocklists.RaiBlocklist{
+				Properties: &raiblocklists.RaiBlocklistProperties{},
+			}
+
+			if model.Description != "" {
+				properties.Properties.Description = &model.Description
+			}
+
+			if _, err := client.CreateOrUpdate(ctx, id, *properties); err != nil {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+
+			metadata.SetID(id)
+			return nil
+		},
+	}
+}
+
+func (c CognitiveRaiBlocklistResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			var model cognitiveRaiBlocklistModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			client := metadata.Client.Cognitive.RaiBlocklistsClient
```

Comment:
Same comment here around the ordering 
```suggestion
			client := metadata.Client.Cognitive.RaiBlocklistsClient
			
			var model cognitiveRaiBlocklistModel
			if err := metadata.Decode(&model); err != nil {
				return fmt.Errorf("decoding: %+v", err)
			}
```